### DISCUSSION TITLE
Disable default UI(Buttons, panels...) to replace it with a custom one

### DISCUSSION THREAD

**Discussion by salah-walid** [iTwin/itwinjs-core#4556] [Q&A] [answered]
Is there a way to disable all UI elements in the Fronstage, meaning all buttons, panels snackbar Etc..., To eventually replace them with our custom UI later ?

**Comment by salah-walid** [comment]
Tried this line in the frontstage developent and it's not changing the theme
```Typescript
UiFramework.registerUserSettingsProvider(new AppUiSettings({colorTheme: "white"}))
```

**Reply by raplemie** [reply]
Can you move this to a separate question so we dont get too much confusion on the original thread?

**Comment by raplemie** [comment]
Hi @salah-walid, there are a couple of options available depending on what is your current starting point and end goal: 
1. If you desire to completely handle the UI on your own, but display a viewport, then you could look at https://github.com/iTwin/viewer-sample-typescript, 
2. If you want to start with an "empty" fronstage, but want to later populate toolbars and widgets, you could use the latest iTwin Viewer and add the following props, which will completly remove the UI but still allow view manipluation with the mouse:
```typescript
<Viewer
  defaultUiConfig={{
    hideNavigationAid: true,
    hideStatusBar: true,
    hideToolSettings: true,
  }}
  uiProviders={[
    new ViewerNavigationToolsProvider({
      horizontal: {},
      vertical: {},
    })
  ]}
  ...otherProps
/>
```
3. If you are already using the AppUi and provide your own frontstage, you can simply not define the relevant section in the frontstage and they will not show up. (`leftPanel, topPanel, rightPanel, bottomPanel, contentManipulationTools, viewNavigationTools, toolSettings, statusBar`)

**Reply by salah-walid** [reply]
Hi @raplemie for the third option where is it possible to "not define" relevant section
I am using StandardFrontstageProvider using StandardFrontstageProps and setting hideNavigationAid to true doesn't hide the navigation buttons

**Reply by NancyMcCallB** [reply]
Hi, @salah-walid ,

The Navigation Aid refers specifically to the device in the corner (e.g., the cube in a spatial view). If you are using the viewer, you can also include the following in the uiconfig:

``` 
navigationTools: {
    hideDefaultHorizontalItems: true,
    hideDefaultVerticalItems: true,
  },
``` 
This uiconfig should hide everything (unless I've missed something):

```
const uiConfig: ItwinViewerUi = {
  contentManipulationTools: {
    cornerItem: {
      hideDefault: true,
    },
    hideDefaultHorizontalItems: true,
    hideDefaultVerticalItems: true,
  },
  navigationTools: {
    hideDefaultHorizontalItems: true,
    hideDefaultVerticalItems: true,
  },
  hideNavigationAid: true,
  hideDefaultStatusBar: true,
  hidePropertyGrid: true,
  hideToolSettings: true,
  hideTreeView: true,
};
```
Thanks,
Nancy McCall

**Reply by salah-walid** [reply]
Hi @NancyMcCallB 
When having a custom frontstage the UIConfig doesn't seem to modify anything.

Thanks

**Reply by NancyMcCallB** [reply]
Correct. The props to the viewer modify the stage it creates for you. If you are using the StandardFrontstageProvider, you'll want to set 
hideNavigationAid, hideToolSettings, and hideStatusBar to true and register a new StandardContentToolsUiItemsProvider and StandardNavigationToolsUiItemsProvider with empty lists for horizontal and vertical toolbars.

**Reply by salah-walid** [reply]
Okay, thanks that worked.
The only thing remaining is how to hide user created panels with UIproviders ?

Thanks

**Reply by NancyMcCallB** [reply]
Which user-created panels? 
If you are using the viewer, simply remove the uiProviders from the Viewer props and you won't get any providers at all. 
After that, you can create your own providers to add the UI you're interested in.

**Comment by NancyMcCallB** [comment, marked as answer]
[my-app.zip](https://github.com/iTwin/itwinjs-core/files/9891520/my-app.zip)

Since the sandboxes are not yet updated to the 3.0 version of the viewer, I made a sample Viewer app that removes everything to illustrate.

**Reply by salah-walid** [reply]
Thank you @NancyMcCallB you are a life saver
